apps/emqx_gateway/src/coap/emqx_coap_api.erl

```diff
@@ -18,47 +18,58 @@
 
 -behaviour(minirest_api).
 
+-include_lib("typerefl/include/types.hrl").
+-include_lib("emqx/include/logger.hrl").
 -include("src/coap/include/emqx_coap.hrl").
 
 %% API
--export([api_spec/0]).
+-export([api_spec/0, paths/0, schema/1, namespace/0]).
 
 -export([request/2]).
 
 -define(PREFIX, "/gateway/coap/clients/:clientid").
--define(DEF_WAIT_TIME, 10).
+-define(DEF_WAIT_TIME, <<"10s">>).
 
--import(emqx_mgmt_util, [ schema/1
-                        , schema/2
-                        , object_schema/1
-                        , object_schema/2
-                        , error_schema/2
-                        , properties/1]).
+-import(hoconsc, [mk/2, enum/1]).
+-import(emqx_dashboard_swagger, [error_codes/2]).
 
 %%--------------------------------------------------------------------
 %%  API
 %%--------------------------------------------------------------------
+namespace() -> "gateway_coap".
+
 api_spec() ->
-    {[request_api()], []}.
+    emqx_dashboard_swagger:spec(?MODULE).
```

Comment:
Suggest: use `emqx_dashboard_swagger:spec(?MODULE, #{check_schema => true, translate_body => true}).`
it will check schema type and translate_body auto.

if the code need binary(), suggest replace string() by binary().
string() and binary() is always string in swagger, but different in erlang.

if we do the above, we don't need convert type in request: Method is already atom.
timeout is the same.
```
ReqType = erlang:binary_to_atom(Method),
{ok, WaitTime} = emqx_schema:to_duration_ms(BinWaitTime),
```

after all this done, we don't crash with 500 if API passes a bad type.

Reply:
thanks, I'll change it